docs/tutorials/using_apis/connectivity/tls_tutorial.md

```diff
@@ -0,0 +1,173 @@
+
+```json
+{
+    "macros": ["MBEDTLS_SHA1_C=1"],
+    "target_overrides": {
+        "*": {
+            "mbed-trace.enable": 1,
+            "tls-socket.debug-level": 0
+        }
+    }
+}
+```
+
+<span class="notes">**Note:** Enabling traces in `mbed_app.json` is not enough to actually show them. You still need to tell the trace library where it needs to write. Calling `mbed_trace_init();` from code writes the traces to the default serial UART port, so you can see them through a serial monitor, the same as when you call `printf`.</span>
+
+## Making an HTTPS request to os.mbed.com
+
+With everything in place, you can now set up a TLS socket connection. On the Mbed website is a file called `hello.txt`, which returns `Hello World!` in plain text.
+
+<span class="notes">**Note:** If you're looking for a fully fledged HTTP/HTTPS library, look at [mbed-http](https://os.mbed.com/teams/sandbox/code/mbed-http/). It also uses TLS Sockets underneath.</span>
+
+### Setting up a TLS socket
+
+<span class="notes">**Note:** There is an [example TLS Socket project](https://github.com/ARMmbed/mbed-os-example-tls-socket) available.</span>
+
+You set up a TLS socket in the same way as you set up a TCP socket, except you call `set_root_ca_cert` with the root CA string.
+
+```cpp
+#include "mbed.h"
+#include "NetworkInterface.h"
+#include <TLSSocket.h>
+
+const char SSL_CA_PEM[] = /* your certificate, see above */ "";
+ 
+int main (void) {
+
+// Get a network interface
+NetworkInterface *network = NetworkInterface::get_default_instance();
+if (network->connect() != 0) {
+    printf("Could not connect to the network...\n");
+    return 1;
+}
+ 
+nsapi_error_t r;
+ 
+// setting up TLS socket
+TLSSocket* socket = new TLSSocket();
+if ((r = socket->open(network)) != NSAPI_ERROR_OK) {
+    printf("TLS socket open failed (%d)\n", r);
+    return 1;
+}
+if ((r = socket->set_root_ca_cert(SSL_CA_PEM)) != NSAPI_ERROR_OK) {
+    printf("TLS socket set_root_ca_cert failed (%d)\n", r);
+    return 1;
+}
+if ((r = socket->connect("os.mbed.com", 443)) != NSAPI_ERROR_OK) {
+    printf("TLS socket connect failed (%d)\n", r);
+    return 1;
+}
+
+}
+```
+
+This now makes the request and returns the content of the file.
+
+```
+Setting up TLS socket...
+[INFO][TLSW]: mbedtls_ssl_conf_ca_chain()
+[INFO][TLSW]: mbedtls_ssl_config_defaults()
+[INFO][TLSW]: mbedtls_ssl_conf_authmode()
+[INFO][TLSW]: mbedtls_ssl_conf_rng()
+[INFO][TLSW]: mbedtls_ssl_setup()
+[INFO][TLSW]: Starting TLS handshake with os.mbed.com
+[INFO][TLSW]: TLS connection to os.mbed.com established
+ 
+[DBG ][TLSW]: Server certificate:
+    cert. version     : 3
+    serial number     : 03:56:D4:79:41:63:31:CA:E0:56:06:61
+    … snip ...
+ 
+[INFO][TLSW]: Certificate verification passed
+```
+
+You can also look at the [complete example](https://github.com/janjongboom/mbed-simulator/blob/6e9f45c61fb10e1983668e777d68e939247b377d/demos/tlssocket/main.cpp) for more details.
```

Comment:
We have a policy that we don't include code from individual user accounts in our public documentation. Do we have example in ARMmbed?

Reply:
As I mentioned above, the content (including code snippets) originates from Jan Jongboom's blog article. We do have a more official [TLS usage code example](https://github.com/ARMmbed/mbed-os-example-tls-socket) and I think I could rewrite this tutorial to use it.